spec/imsc-hrm.html

```diff
@@ -449,40 +449,41 @@ <h2>Architecture</h2>
       <h2>General</h2>
 
       <p>The Presentation Compositor SHALL render in Presentation Buffer P<sub>n</sub> each successive <a>Intermediate Synchronic
-      Document</a> E<sub>n</sub> using the following steps in order:</p>
+      Document</a> E<sub>n</sub> that is not an <a>empty ISD</a> using the following steps in order:</p>
 
       <ol>
-        <li>clear the pixels of the entire <a>Root Container Region</a>, unless n=0 or E<sub>n-1</sub> is an <a>empty ISD</a>;</li>
-
         <li>paint, according to stacking order, all background pixels for each region;</li>
 
-        <li>paint all pixels for background colors associated with text or image subtitle content; and</li>
+        <li>paint all pixels for background colors associated with text or image subtitle content;</li>
+
+        <li>paint the text or image subtitle content; and</li>
 
-        <li>paint the text or image subtitle content.</li>
+        <li>clear the pixels of the entire <a>Root Container Region</a>.</li>
       </ol>
 
       <p>The Presentation Compositor SHALL start rendering E<sub>n</sub>:</p>
 
       <ul>
-        <li>at the presentation time of E<sub>m</sub>, where E<sub>m</sub> is the closest ISD for which 0&lt;m&lt;n and E<sub>m-1</sub>
+        <li>at the presentation time of E<sub>m</sub>, where E<sub>m</sub> is the closest ISD for which 0&lt;m&lt;n and E<sub>m</sub>
         is not an <a>empty ISD</a>, if the presentation time of E<sub>n</sub> minus that of E<sub>m</sub> is less than <a>IPD</a>; or
       </li>
         
         <li>at the presentation time of E<sub>n</sub> minus <a>IPD</a>, otherwise.</li>
       </ul>
 
-      <p class="note">The Presentation Compositor never renders an ISD more than <a>IPD</a> ahead of its presentation time and treats sequences of
-        <a>empty ISDs</a> as a single ISD.</p>
+      <p class="note">The Presentation Compositor never renders an ISD more than <a>IPD</a> ahead of its presentation
+      time and effectively handles a sequence of <a>empty ISDs</a> as part of the previous ISD that is not an <a>empty
+      ISD</a>.</p>
 
       <figure id='fig-rendering-presentation-time'>
         <img src="figures/isd-rendering-time-fig.png" alt="ISD rendering and presentation times.">
```

Comment:
I'm finding this diagram quite confusing. It looks like a non-empty ISD begins to be rendered at the _end_ of the presentation time of the previous one, and doesn't make clear when presentation time is vs rendering time.

Reply:
#52 raised.